app/src/main/java/com/example/music_app/domain/repositories/LoginRepository.kt

```
@@ -1,14 +1,14 @@
 package com.example.music_app.domain.repositories
 
 import com.example.music_app.data.models.ResponseError
+import com.example.music_app.data.models.TokenRefreshResponse
 import com.example.music_app.data.models.TokenResponse
 import com.github.michaelbull.result.Result
 import kotlinx.coroutines.flow.Flow
 
 interface LoginRepository {
     suspend fun requestToken(code: String): Flow<Result<TokenResponse, ResponseError>>
-
-    suspend fun saveToken(token: String)
-
-    fun isAuthorized(): Flow<Boolean>
+    suspend fun requestRefreshToken(): Flow<Result<TokenRefreshResponse, ResponseError>>
+    suspend fun isAuthorized(): Result<Boolean, Throwable>
+    suspend fun isOutdated(): Result<Boolean, Throwable>
```

Instead of `Throwable` It would be better to return a custom Error, Throwables should be parsed on the data layer a returned as a CustomErrors with a message.